CRM/Contact/BAO/Group.php

```diff
@@ -1016,15 +1016,16 @@ public static function getGroupList(&$params) {
         $values[$object->id]['created_by'] = "<a href='{$contactUrl}'>{$object->created_by}</a>";
       }
 
-      // get group contact count using Contact.GetCount API
-      // CRM-20226 This has been added here in order to address a specific bug. However, a prior
-      // decision was to refresh group counts less aggressively, offering instead a button to
-      // refresh them to give users a better experience by loading pages quicker.
-      // For some sites this can be crazy slow even though only 25 sites resolve. Even for sites
-      // with relatively few smart groups it is not a good user experience.
-      // Adding comments here as I'm not going to tackle a fix this time around but want
-      // to warn people off making it worse.
-      $values[$object->id]['count'] = civicrm_api3('Contact', 'getcount', array('group' => $object->id));
+      // By default, we try to get a count of the contacts in each group
+      // to display to the user on the Manage Group page. However, if
+      // that will result in the cache being regenerated, then dipslay
+      // "unknown" instead to avoid a long wait for the user.
+      if (CRM_Contact_BAO_GroupContactCache::shouldGroupBeRefreshed($object->id)) {
+        $values[$object->id]['count'] = ts('unknown');
```

Comment:
Ahh I see, happy with this final change :) One last thing, should we place any helpicon beside 'Count' title to notify the user about the ```unknown``` possibilities?

@

Reply:
that would be nice I expect. Ideally you would be able to refresh on a per group basis - although I don't think that needs to block